web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -323,23 +330,28 @@ def _make_f_expression(field, location_path):
     return F("{}__{}".format(location_path, field))
 
 
-def _date_range(date_spec=None):
+def _date_range(calendar_span, focal_date=None):
     # returns (start_date, end_date)
     # start_date is the latest monday that is less than or equal to today,
     # end_date is start_date + seven days
     # both values are then padded by 24 hours to allow for TZ differences
 
-    # throws ValueError if date_spec is not in ISO-8601 format
+    # throws ValueError if focal_date is not in ISO-8601 format
 
-    if date_spec:
-        initial_date = datetime.strptime(date_spec, ISO_8601_DATE_FORMAT)
+    if focal_date:
+        initial_date = datetime.strptime(focal_date, ISO_8601_DATE_FORMAT)
     else:
         initial_date = datetime.now()
     initial_date = utc.localize(initial_date)
     # This calculation depends on the fact that monday == 0 in python
-    start_date = initial_date - timedelta(initial_date.weekday())
-    end_date = start_date + ONE_WEEK + ONE_DAY
-    adjusted_start_date = start_date - ONE_DAY
+    if calendar_span == "monthly":
+        start_date = initial_date - timedelta(int(initial_date.strftime("%d")))
```

Comment:
This computation differs from the one for "week" only in two numbers, the stepback number that you use to compute the start of the range and the span. Everything else about the computation, including the padding by a day, is the same. 

You could do:
```
if span = "month":
    step_back = timedelta(int(initial_date.strftime("%d")))
    span = ONE_MONTH
else:
    step_back = timedelta(initial_date.weekday())
    span = ONE_WEEK
return compute_date_range(initial_date, step_back, range)
```

Reply:
The similarity is just one line `(the end_date line)`  `adjusted_start_date` calculation is different for both

Reply:
Okay, we should probably avoid this pattern in the future but I'm going to let it go for now.